docs/guide/request_lifecycle.md

```diff
@@ -0,0 +1,155 @@
+
+
+### Routing
+
+Lately, also listening the `kernel.request` event, the Symfony\Component\HttpKernel\EventListener\RouterListener (priority 32) calls the eZ\Publish\Core\MVC\Symfony\Routing\ChainRouter::matchRequest and add its returned parameters to the Request.
+
+ChainRouter:
+The [ChainRouter](https://symfony.com/doc/current/cmf/components/routing/chain.html) is a Symfony Content Management Framework (CMF) component.
+It has a collection of prioritized routers to crawl in until finding one matching the Request to define what to do next.
+The ChainRouter router collection is build by the ChainRoutingPass collecting the services tagged `router` (see `bin/console debug:container --tag=router;`).
+The DefaultRouter is always added to the collection with top priority.
+
+DefaultRouter:
+The DefaultRouter (see `bin/console debug:container router.default;`) is trying to match the `semanticPathinfo` against routes, close to [the way pure Symfony does](https://symfony.com/doc/current/routing.html) by using the Symfony\Component\Routing\Router.
+If a route matches, the controller associated to it will have the responsibility to build a Response object.
+
+UrlWildcardRouter (not on the schema):
+If [URL Wildcards](https://doc.ibexa.co/en/latest/guide/url_management/#url-wildcards) have been enabled, then the URLWildcardRouter (see `bin/console debug:container ezpublish.urlwildcard_router;`) is the next tried router.
+If a wildcard matches, the Request's `semanticPathinfo` is updated and the router pretend a ResourceNotFoundException to continue with the ChainRouter collection's next entry.
+
+UrlAliasRouter:
+This router will use the UrlAliasService to associate the `semanticPathinfo` to a location.
+If a location is found, the Request will receive the attributes `locationId` and `contentId` as well as that `viewType` is `full` and that, for now, the `_controller` is `ez_content:viewAction` (see `bin/console debug:container ez_content;`).
+
+*Notice about Permission Control*: Another `kernel.request` event listener is the EzSystems\EzPlatformAdminUi\EventListener\RequestListener (priority 13): When a route got a `siteaccess_group_whitelist` parameter, this listener check that the current siteaccess is in one of the listed groups. For example, the Admin UI set an early protection of its route by passing them a `siteaccess_group_whitelist` containing only the `admin_group`.
+[TODO: https://doc.ibexa.co/en/latest/guide/permissions/#permissions-for-routes]
+
+Now, the Request know its controller, the HttpKernel dispatch the `kernel.controller` event.
+
+
+
+## Kernel's Controller Event
+
+(see `bin/console debug:event-dispatcher kernel.controller;`)
```

Comment:
I'm wondering in what situation a user may need this information? Is there a use case for it?

Anyway, if there is, I'd suggest something like:

```suggestion
To view all listeners that listen to this event, run `php bin/console debug:event-dispatcher kernel.controller`.
```

Reply:
@DominikaK I didn't know if this level of detail was relevant. As a developer, I found observing the result of this command instructive. But it also sounds like asking to the reader to document himself. I'll probably resume all those invitations to execute a command to only one `!!!tip` for people who want to know more about events and services cited in the article.

Reply:
Yep, one tip sounds like a good idea.